skuba-update/VERSION

```diff
@@ -0,0 +1 @@
+4.3.0
```

Comment:
> No, this is wrong. It radiacally changes skuba-update versioning numbers, current released version is `1.2.9`.

Yes I changed the version number, version tbd. I can switch to something else
Currently it's 0.0.0 on my machine. It started to break when we removed the version file here https://github.com/SUSE/skuba/commit/5c36f8d031e956d7a332e14ef580626288670d77. And the wrong thing, is that we assume an empty string if it can't find the version file. It should have failed instead. This fixes it, as explained in the commit message.

I have to figure out where you found your version file so that it gives you `1.2.9` when installing with the rpm.

> 
> Also, it has to be the same version as the one used in the spec file. So we can't hardcode the version here, it needs to be in sync with skuba version.

I disagree there. I don't see a reason to sync with skuba. This is different code, different purposes. Hardcoding is a matter of preference, and it used to be that way, and working fine. It's also very simple to release.

> 
> To go that way we need to split the skuba-update into its own source RPM, thus release `skuba` and `skuba-update` as separate artifacts and, for sanity, remove if from skuba github repository and create its own. All out of scope of this PR I'd say.
> 
> Also, AFAIK, the `skuba-update --version` works fine if you have the RPM installed.
> Customers currently see:
> 
> ```
> $ skuba-update --version
> skuba-update 1.2.9
> ```
> 
> This is still working fine at build time however I just noticed that [5c36f8d](https://github.com/SUSE/skuba/commit/5c36f8d031e956d7a332e14ef580626288670d77) broke the `--version` flag. It was partially fixed on [811cb90](https://github.com/SUSE/skuba/commit/811cb9051ca4f5238ffc7f1f6403ed14346d5c44), as this does not solve the development environment case.
> 
> Regarding the `--version` flag I'd say we can either leave it as is (since this is just an issue for the development environment) or make the version methods aware of the changes introduced in [5c36f8d](https://github.com/SUSE/skuba/commit/5c36f8d031e956d7a332e14ef580626288670d77) and provide a better solution than [811cb90](https://github.com/SUSE/skuba/commit/811cb9051ca4f5238ffc7f1f6403ed14346d5c44).

Reply:
For now, I vote keeping the version number of skuba and skuba_update in sync with the tag. I see there was a problem in https://github.com/SUSE/skuba/commit/5c36f8d031e956d7a332e14ef580626288670d77 that if the checked out commit didn't match the commit id of one of the tags then an empty result was returned, so we need a method that is better at determining the correct, current tag version (or allows a developer to override the value for upgrade/downgrade testing).